test/fiddle/test_pointer.rb

```diff
@@ -161,11 +161,16 @@ def test_to_ptr_with_ptr
       end
     end
 
-    def test_to_ptr_with_num
+    def test_to_ptr_with_int
       ptr = Pointer.new 0
       assert_equal ptr, Pointer[0]
     end
 
+    def test_to_ptr_with_num
+      ptr = Pointer.new 0
+      assert_equal ptr, Pointer[0.0]
```

Comment:
The only reason I added this test is to make sure the functionality of the FFI backend matches the C implementation. I just picked an arbitrary builtin that can be coerced into an integer 🤷🏻‍♀️  There might be other types that make more sense, or I could just create a dummy int-wrapper and use that instead, e.g.

```ruby
IntWrapper = Struct.new(value) do
  def to_int
    value
  end
end
```
and then
```suggestion
      assert_equal ptr, Pointer[IntWrapper.new(0)]
```

Reply:
Ah, this is just a test for coercion, I guess.  @danini-the-panini maybe adding the wrapper will show the intention of the test better?  I don't really have a strong opinion about float vs wrapper object as long as the intention of the test is clear.  Either a comment or the wrapper class seems sufficient to me.

I _assume_ this is calling `to_int` on things because it wants to accept other pointer instances as a parameter.

Reply:
It's calling `rb_Integer` in C, and `Kernel#Integer` in the FFI backend

Reply:
@kou @tenderlove I've updated it to use a wrapper and renamed the test method

Reply:
> It's calling `rb_Integer` in C, and `Kernel#Integer` in the FFI backend

ah, gotcha.

> @kou @tenderlove I've updated it to use a wrapper and renamed the test method

Thank you!